com.ibm.wala.cast.js.rhino/src/main/java/com/ibm/wala/cast/js/translator/RhinoToAstTranslator.java

```diff
@@ -1225,6 +1225,13 @@ private String getParentName(AstNode fn) {
 
     @Override
     public CAstNode visitFunctionNode(FunctionNode fn, WalkContext context) {
+      // Declare a unique "this" bind to the current context's this
+      String thisName = "this_" + context.top().hashCode();
```

Comment:
It would be cleaner if we only created this variable when we needed it (i.e., when there is a nested arrow function).  But if that's tricky, as a first cut I am ok with creating it always.

Reply:
Right, I will leave a FIXME and try to address it after the initial validation